### PR TITLE
[HLS-3363] collect data and propagate context

### PR DESCRIPTION
**Pull request by nozik**

### CODE REVIEW

File: proxy-libs/heliossqs/heliossqs.go [hunk 1]
```diff
@@ -28,27 +34,115 @@ const ServiceAPIVersion = origin_sqs.ServiceAPIVersion
...
+			}
+
+			if len(attrs) < maxSqsAttributesCount {
+				carrier := sqsMessageCarrier{attrs}
+				otel.GetTextMapPropagator().Inject(ctx, carrier)
+			}
+
+			result = append(result, attribute.KeyValue{Key: "messaging.payload", Value: attribute.StringValue(*castParams.MessageBody)})
+		}
+	case *origin_sqs.SendMessageBatchInput:
```
Comment (LironKS) [inline proxy-libs/heliossqs/heliossqs.go:89]: I don't understand the other 2 cases here - what's the purpose of them?

Comment (nozik) [reply proxy-libs/heliossqs/heliossqs.go:89]: We want to instrument 3 different types of sqs operations:

1. Send a single message (we want to inject context and collect data)
2. Send multiple message (same as #1)
3. We are received a message we need to make sure that the `traceparent` attribute is collected, as it isn't done by default)

File: proxy-libs/heliossqs/heliossqs.go [hunk 2]
```diff
@@ -28,27 +34,115 @@ const ServiceAPIVersion = origin_sqs.ServiceAPIVersion
...
+					attrs = map[string]types.MessageAttributeValue{}
+					entry.MessageAttributes = attrs
+				}
+
+				if len(attrs) < maxSqsAttributesCount {
+					carrier := sqsMessageCarrier{attrs}
+					otel.GetTextMapPropagator().Inject(ctx, carrier)
+				}
+
+				carrier := sqsMessageCarrier{attrs}
```
Comment (LironKS) [inline proxy-libs/heliossqs/heliossqs.go:105]: you dont need an else here?

Comment (nozik) [reply proxy-libs/heliossqs/heliossqs.go]: Good catch, fixed

File: proxy-libs/heliossqs/heliossqs.go (hunk 1 above)
Comment (LironKS) [reply proxy-libs/heliossqs/heliossqs.go:89]: why you aren't collecting data in case #2?

Comment (nozik) [reply proxy-libs/heliossqs/heliossqs.go:90]: Done

**Review by LironKS — APPROVED** [review]

**Pull request merged**